adapters/yssp/yssp.go

```diff
@@ -0,0 +1,213 @@
+package yssp
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type YSSPAdapter struct {
+	URI string
+}
+
+func (a *YSSPAdapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	errors := make([]error, 0, 1)
+
+	if len(request.Imp) == 0 {
+		err := &errortypes.BadInput{
+			Message: "No impression in the bid request",
+		}
+		errors = append(errors, err)
+		return nil, errors
+	}
```

Comment:
Please remove this duplicate check. The core logic of PBS ensures adapters are not called without at least one impression.